tensorrt/inference.cu

```diff
@@ -75,15 +105,15 @@ auto Inference::createCudaEngine(std::filesystem::path const& onnxModelPath, std
 
         // Save Engine to File
         auto trtModelStream = tempEng->serialize();
-        std::ofstream outputFileStream{enginePath, std::ios::binary};
-        outputFileStream.write(static_cast<char const*>(trtModelStream->data()), trtModelStream->size());
+        std::ofstream outputFileStream{mEngineModelPath, std::ios::binary};
+        outputFileStream.write(static_cast<char const*>(trtModelStream->data()), static_cast<int32_t>(trtModelStream->size()));
```

Comment:
why int32 cast?

Reply:
narrowing conversion, but I made it more clear